src/wfs/src/main/java/org/geoserver/wfs/response/v2_0/HitsOutputFormat.java

```diff
@@ -34,8 +36,36 @@ public boolean canHandle(Operation operation) {
         return request != null && request.getResultType() == ResultTypeType.HITS;
     }
 
+    public String getMimeType(Object value, Operation operation) throws ServiceException {
```

Comment:
Output formats are normally format specific (with a few exceptions, like XSLT or OGR based format generators, where the format is decided at runtime). Can you roll a superclass and have two small subclasses handling the details of GML vs JSON encoding instead?